crates/iceberg-catalog/migrations/20240898134304_table_as_tables.sql

```diff
@@ -0,0 +1,232 @@
+-- Add migration script here
+create type table_format_version as enum ('1', '2');
+
+alter table "table"
+    add column table_format_version table_format_version;
+
+UPDATE "table"
+SET table_format_version = (metadata ->> 'format-version')::table_format_version
+WHERE table_format_version IS NULL;
+
+alter table "table"
+    alter column table_format_version set not null;
+
+create table table_schema
+(
+    schema_id int   not null,
+    table_id  uuid  not null REFERENCES "table" (table_id) ON DELETE CASCADE,
+    schema    jsonb not null,
+    CONSTRAINT "unique_schema_per_table" unique (table_id, schema_id),
+    PRIMARY KEY (table_id, schema_id)
+);
+
+call add_time_columns('table_schema');
+select trigger_updated_at('table_schema');
+
+INSERT INTO table_schema (schema_id, table_id, schema)
+SELECT (schema ->> 'schema-id')::int, table_id, schema
+FROM "table",
+     jsonb_array_elements(metadata -> 'schemas') AS schema;
+
+create table table_current_schema
```

Comment:
I don't think we should have separate tables for all the _current and _default (sort order, schema and partition spec). Insstead we just have the current_id in the main table in a column and all schemas for a table in another with the current_id beeing a foreign key.